src/Functions/PositionImpl.h

```diff
@@ -198,6 +198,48 @@ struct PositionImpl
         const UInt8 * const end = haystack_data.data() + haystack_data.size();
         const UInt8 * pos = begin;
 
+        /// Fastpath when needle is empty
```

Comment:
[Docs of `position()`](https://clickhouse.com/docs/en/sql-reference/functions/string-search-functions/#positionhaystack-needle-locatehaystack-needle) don't mention what the result is if the needle is empty. Maybe you could add a sentence or two.